src/ui/components/map/providers/mapboxmapprovider.js

```diff
@@ -3,45 +3,71 @@
 import MapProvider from './mapprovider';
 import DOM from '../../../dom/dom';
 
-/* global mapboxgl */
+/* global mapboxgl, MapboxLanguage */
 
 /**
  * MapBoxMapProvider is an implementation of a MapProvider
  * that handles the integration with the third party API to expose maps.
  * @extends MapProvider
  */
 export default class MapBoxMapProvider extends MapProvider {
+  constructor (opts = {}, systemOpts = {}) {
+    super(opts, systemOpts);
+
+    this._isMapboxLoaded = false;
+    this._isMapboxLanguageLoaded = false;
+  }
   /**
    * Load the external JS Library
    * @param {function} onLoad An optional callback to invoke once the JS is loaded.
    */
   loadJS (onLoad) {
-    let script = DOM.createEl('script', {
+    let mapboxScript = DOM.createEl('script', {
       id: 'yext-map-js',
       onload: () => {
-        this._isLoaded = true;
-        mapboxgl.accessToken = this._apiKey;
-
-        if (typeof onLoad === 'function') {
-          onLoad();
-        }
-
-        if (typeof this._onLoaded === 'function') {
-          this._onLoaded();
-        }
+        this._isMapboxLoaded = true;
+        this._onScriptLoad(onLoad);
       },
       async: true,
       src: 'https://api.mapbox.com/mapbox-gl-js/v0.44.1/mapbox-gl.js'
     });
 
+    let mapboxLanguageScript = DOM.createEl('script', {
+      id: 'yext-map-language-js',
+      onload: () => {
+        this._isMapboxLanguageLoaded = true;
+        this._onScriptLoad(onLoad);
+      },
+      async: true,
+      src: 'https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-language/v0.10.1/mapbox-gl-language.js'
+    });
+
     let css = DOM.createEl('link', {
       id: 'yext-map-css',
       rel: 'stylesheet',
       href: 'https://api.mapbox.com/mapbox-gl-js/v0.44.1/mapbox-gl.css'
     });
 
     DOM.append('body', css);
-    DOM.append('body', script);
+    DOM.append('body', mapboxScript);
+    DOM.append('body', mapboxLanguageScript);
+  }
+
+  _onScriptLoad (onLoadCB) {
+    if (!this._isMapboxLoaded || !this._isMapboxLanguageLoaded) {
+      return;
+    }
+
+    this._isLoaded = true;
+    mapboxgl.accessToken = this._apiKey;
+
+    if (typeof onLoad === 'function') {
+      onLoadCB();
```

Comment:
Do we need to do any kind of `this` binding before passing `onLoadCB` to `_onScriptLoad`?

Reply:
Possibly, however since implementing Oliver's suggestion of bundling the MapboxLanguage script, this is now outdated